crates/re_viewer/src/viewer_analytics.rs

```diff
@@ -61,29 +69,40 @@ impl ViewerAnalytics {
         use crate::AppEnvironment;
         let app_env_str = match app_env {
             AppEnvironment::PythonSdk(_) => "python_sdk",
-            AppEnvironment::RustSdk => "rust_sdk",
+            AppEnvironment::RustSdk(_) => "rust_sdk",
             AppEnvironment::RerunCli => "rerun_cli",
             AppEnvironment::Web => "web",
         };
         self.register("app_env", app_env_str.to_owned());
 
-        if let AppEnvironment::PythonSdk(version) = app_env {
-            self.register("python_version", version.to_string());
-        }
-
         #[cfg(all(not(target_arch = "wasm32"), feature = "analytics"))]
         if let Some(analytics) = &self.analytics {
             let rerun_version = env!("CARGO_PKG_VERSION");
-            let rust_version = env!("CARGO_PKG_RUST_VERSION");
             let target = re_analytics::TARGET_TRIPLET;
             let git_hash = re_analytics::GIT_HASH;
 
             let mut event = Event::update("update_metadata".into())
                 .with_prop("rerun_version".into(), rerun_version.to_owned())
-                .with_prop("rust_version".into(), rust_version.to_owned())
                 .with_prop("target".into(), target.to_owned())
                 .with_prop("git_hash".into(), git_hash.to_owned());
 
+            // If we happen to know the Python or Rust version used on the _host machine_, i.e. the
+            // machine running the viewer, then add it to the permanent user profile.
+            //
+            // The Python/Rust versions appearing in user profiles always apply to the host
+            // environment, _not_ the environment in which the data logging is taking place!
+            if let AppEnvironment::RustSdk(version) = &app_env {
+                event = event.with_prop("rust_version".into(), version.clone());
+            } else {
+                event = event.with_prop(
+                    "rust_version".into(),
+                    env!("CARGO_PKG_RUST_VERSION").to_owned(),
+                );
+            }
```

Comment:
We are in the `re_viewer` crate here - how could the version in `AppEnvironment::RustSdk` ever differ from `CARGO_PKG_RUST_VERSION`? Seems like it is the same thing, just with a lot more steps?

Reply:
It can't, I just wanted the code to be perfectly symmetric on both sides to reflect that the versions that appear in `AppEnvironment` vs. `RecordingSource` are in fact two sides of the same coin basically.

I still think the symmetry in code is nice for comprehension, though I agree this if/else statement is quite fugly... The only reason it is there is because `RerunCli` doesn't carry its version around right now, but it should I think.